From 6236136418a437b5dc2d11c9c83b6e74f246aece Mon Sep 17 00:00:00 2001
From: Alberto Mardegan <mardy@users.sourceforge.net>
Date: Wed, 26 Apr 2023 21:05:37 +0300
Subject: [PATCH 3/8] Lib: do not attempt to terminate the GTask twice

Move the "finish" label after the calls to g_task_return_*() since we
already called g_task_return_error_if_cancelled() in case the job was
cancelled.

Fixes: #11
---
 libaccounts-glib/ag-manager.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libaccounts-glib/ag-manager.c b/libaccounts-glib/ag-manager.c
index 5460b0b..a9136e6 100644
--- a/libaccounts-glib/ag-manager.c
+++ b/libaccounts-glib/ag-manager.c
@@ -1108,7 +1108,6 @@ exec_transaction_idle (StoreCbData *sd)
                                      "Generic error");
     }
 
-finish:
     if (error != NULL)
     {
         g_task_return_error (sd->task, error);
@@ -1118,6 +1117,7 @@ finish:
         g_task_return_boolean (sd->task, TRUE);
     }
 
+finish:
     _ag_account_store_completed (account, sd->changes);
 
     priv->locks = g_list_remove (priv->locks, sd);
-- 
2.44.0

